lib/spack/env/cc

```diff
@@ -124,16 +135,30 @@ prepend() {
 # assuming the list is separated by SEP.
 # Handles empty lists and single-element lists.
 append() {
-    varname="$1"
-    elt="$2"
-
-    if empty "$varname"; then
-        eval "$varname=\"\${elt}\""
+    if [ -z "$3" ]; then
+        setsep "$1"
     else
-        # Get the appropriate separator for the list we're appending to.
-        setsep "$varname"
-        eval "$varname=\"\${$varname}${sep}\${elt}\""
+        sep="$3"
     fi
+    eval "$1=\"\${$1:+\$$1${sep}}\${2}\""
+}
+
+# appends LISTNAME ELEMENT SEP
+#
+# Append ELEMENT to the list stored in the variable LISTNAME,
+# assuming the list is separated by SEP.
+# Handles empty lists and single-element lists.
+appends() {
+    eval "$1=\"\${$1:+\$$1${3}}\${2}\""
+}
+
+# appends LISTNAME ELEMENTVAR SEP
+#
+# Append ELEMENT to the list stored in the variable LISTNAME,
+# assuming the list is separated by SEP.
+# Handles empty lists and single-element lists.
+appendsv() {
```

Comment:
It looks like this should be `appendv` to match other methods. The documentation comment also needs to make clear how this differs from `append`, and that $ELEMENTVAR is getting appended.

Reply:
Yeah, I'll update this doc comment significantly.  To sort it out in my own head and have it written down:

* append: append string ELEMENT to list named LISTNAME using SEP if specified and otherwise determine it from LISTNAME
* appends: the s suffix means the SEP argument is *required* and is not checked, in the vast majority of cases we know the type of the list when calling it, and setsep is surprisingly expensive
* appendsv: required SEP argument and internal dereference of ELEMENTVAR to retrieve ELEMENT (v)